### PR TITLE
Fix list machine ids

### PR DESCRIPTION
**Pull request by ptzianos**
Fix for issue <https://github.com/mistio/mist.core/issues/1869>
Set machine ids from db to the dicts returned to the frontend when list_machines is invoked.

---
This change is [Reviewable](https://reviewable.io/reviews/mistio/mist.io/772)

Also changes Machine model to also have last_seen and missing_since values that can be used later on to handle missing machines.

### CODE REVIEW

**Comment by ptzianos** [comment]
Run tests

File: src/mist/io/methods.py [hunk 1]
```diff
@@ -1483,26 +1483,40 @@ def get_machine_actions(machine_from_api, conn, extra):
...
+                        for machine in Machine.objects(cloud=cloud)}
+    now = datetime.utcnow()
     ret = []
-    for m in machines:
+
+    for m in machines_from_provider:
+
+        machine_entry = machines_from_db.pop(m.id, None)
+        if machine_entry:
+            machine_entry.update(set__last_seen=now, set__missing_since=None)
```
Comment (dimrozakis) [inline src/mist/io/methods.py]: Why not just set the attributes and the call `.save()`? Doing this using `update` and `set__` atomic operations is not really that more efficient (you have already loaded the machine model instance) and is harder to read. Also, atomic operations such as this bypass any type validation/sanitization that the model usually provides, as we found out the bitter way with the notorious bug we had concerning machine id's.

So, I'm not saying that this is wrong, but why not keep it simple?

File: src/mist/io/methods.py [hunk 2]
```diff
@@ -1601,20 +1607,31 @@ def list_machines(user, cloud_id):
...
+                   'last_seen': str(machine_entry.last_seen)
+                        if machine_entry and machine_entry.last_seen else '',
                    'extra': m.extra}
         machine.update(get_machine_actions(m, conn, m.extra))
         ret.append(machine)
     if conn.type == 'libvirt':
         # close connection with libvirt
         conn.disconnect()
+
+    # import rpdb; rpdb.set_trace()
```
Comment (dimrozakis) [inline src/mist/io/methods.py]: Please remove this

File: src/mist/io/methods.py [hunk 3]
```diff
@@ -1601,20 +1607,31 @@ def list_machines(user, cloud_id):
...
+
+    # mark machines that are no longer available in list_nodes as missing
+    for machine_entry in machines_from_db.values():
+        if not machine_entry.missing_since:
+            machine_entry.update(set__missing_since=now)
```
Comment (dimrozakis) [inline src/mist/io/methods.py]: Same comment as above concerning update vs save.

**Comment by dimrozakis** [comment]
Run tests

**Comment by d-mo** [comment]
Run tests

**Pull request merged**